criu/sk-inet.c

```diff
@@ -402,6 +402,8 @@ static int dump_ip_opts(int sk, int family, int type, int proto, IpOptsEntry *io
 		/* Due to kernel code we can use SOL_IP instead of SOL_IPV6 */
 		ret |= dump_opt(sk, SOL_IP, IP_FREEBIND, &ioe->freebind);
 		ioe->has_freebind = ioe->freebind;
+		ret |= dump_opt(sk, SOL_IP, IP_PKTINFO, &ioe->pktinfo);
```

Comment:
Would this also work with `IPV6_PKTINFO`?

Reply:
Kernel code for IP(V6)_PKTINFO seem completely different unlike IP_FREEBIND:
```
case IP_PKTINFO:
                if (val)
                        inet->cmsg_flags |= IP_CMSG_PKTINFO;
                else
                        inet->cmsg_flags &= ~IP_CMSG_PKTINFO;
```
```
case IPV6_PKTINFO:
        {       
                struct in6_pktinfo pkt;
                                
                if (optlen == 0)
                        goto e_inval;
                else if (optlen < sizeof(struct in6_pktinfo) ||
                         sockptr_is_null(optval))
                        goto e_inval;

                if (copy_from_sockptr(&pkt, optval, sizeof(pkt))) {
                        retv = -EFAULT;
                        break;
                }
                if (!sk_dev_equal_l3scope(sk, pkt.ipi6_ifindex))
                        goto e_inval;

                np->sticky_pktinfo.ipi6_ifindex = pkt.ipi6_ifindex;
                np->sticky_pktinfo.ipi6_addr = pkt.ipi6_addr;
                retv = 0;
                break;
        }
```
So it seems I should have different paths for ipv6 and ipv4 here -> will try to do.

Reply:
It seems that IPV6_PKTINFO is a completely different thing, systemd uses IPV6_RECVPKTINFO for ipv6 similar to IP_PKTINFO.

Reply:
done.